code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -525,27 +526,29 @@
 	else if(istype(W, /obj/item/borg/upgrade/))
 		var/obj/item/borg/upgrade/U = W
 		if(!opened)
-			usr << "<span class='warning'>You must access the borgs internals!</span>"
+			user << "<span class='warning'>You must access the borgs internals!</span>"
 		else if(!src.module && U.require_module)
-			usr << "<span class='warning'>The borg must choose a module before it can be upgraded!</span>"
+			user << "<span class='warning'>The borg must choose a module before it can be upgraded!</span>"
 		else if(U.locked)
-			usr << "<span class='warning'>The upgrade is locked and cannot be used yet!</span>"
+			user << "<span class='warning'>The upgrade is locked and cannot be used yet!</span>"
 		else
 			if(U.action(src))
-				usr << "<span class='notice'>You apply the upgrade to [src].</span>"
-				usr.drop_item()
+				if(!user.drop_item())
+					return

```

Comment:
The check must be done before U.action is called, otherwise the upgrade will be done even if the object can't be dropped.